### DISCUSSION TITLE
Help Building a 3-level Cache (Private L1, Private L2, Shared L3 Hierarchy)

### DISCUSSION THREAD

**Discussion by ghost** [gem5/gem5#969] [Q&A]
Hello! I'm trying to expand on the private_l1_private_l2 and private_l1_shared_l2 cache hierarchy classes already included in the source Python for gem5 by building a three level cache that uses a private_l1_private_l2_shared_l3 hierarchy. So far, I've done so creating an L3Cache class and a PrivateL1PrivateL2SharedL3Hierarchy class - pretty much following the same template as the pre-existing models. I'm wondering if the way I'm going about creating the bus between the L2 caches and the shared L3 cache is correct. In my code, I'm just using the same L2XBar() on line 132 that is used in both two-level cache classes that are already implemented. I already have one connecting each private l1 cache to each private l2 cache. Should I be using something else for the bus connecting the shared L3 cache (i.e., do I need to make an L3XBar()?)
![Screenshot 2024-03-27 at 1 38 42 AM](https://github.com/gem5/gem5/assets/90482410/5f22e435-36f4-408d-9de5-d1cf392aacc4)

**Comment by qqli579** [comment]
If the latency of your L3Xbar and the latency of L2Xbar are the same, you can use L2Xbar. If they are different, you can To create a separate L3Xbar
